ldapauthenticator/ldapauthenticator.py

```diff
@@ -335,10 +336,7 @@ def resolve_username(self, username_supplied_by_user):
             userdn=self.lookup_dn_search_user,
             password=self.lookup_dn_search_password,
         )
-        if not conn.bind():
-            self.log.warning(
-                f"Failed to connect to LDAP server with search user '{self.lookup_dn_search_user}'"
-            )
+        if not conn:
```

Comment:
This is right after calling get_connection, which always does a bind operation because we have auto_bind set to a value making it so. This makes the `if not conn.bind()` too much and it should at most be checking `conn.bound` property.

This PR updated the get_connection to return none and log a warning if the bind operation failed though, so we now just check if conn is set truthy or not going onwards here.